### DISCUSSION TITLE
which parts can be learned to help build a simple backtesting system？

### DISCUSSION THREAD

**Discussion by EisbergDu** [nkaz001/hftbacktest#154] [General]
I got some signals by myself through some machine training models, now I just want to do a fast and accurate backtesting of these signals, but it's slightly redundant if I use the whole system, which parts I can learn from to build a fast backtesting framework? Thanks for your help!

**Comment by nkaz001** [comment]
If your signal execution is not sensitive - such as when dealing with MFT or LFT - it may be more efficient to use another backtesting framework based on bars, as these frameworks are faster than hftbacktest. But, if precise fill simulation is critical, such as in HFT signals, you will need to use a tick-by-tick backtesting approach with queue position modeling, which is good to use hftbacktest.

Note that you don't need to calculate the signal during backtesting; instead, you can pre-compute the signal with its corresponding timestamps and use it directly within the hftbacktest framework for execution only. For more details, see the [example](https://github.com/nkaz001/hftbacktest/blob/master/examples/Integrating%20Custom%20Data.ipynb).

**Reply by EisbergDu** [reply]
thanks!
